tests/framework/extensions/kubectl/template.go

```diff
@@ -0,0 +1,61 @@
+package kubectl
+
+import (
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+)
+
+const (
+	// rancherShellSettingID is the setting ID that used to grab rancher/shell image
+	rancherShellSettingID = "shell-image"
+	// kubeConfig is a basic kubeconfig that uses the pod's service account
+	kube = `
+apiVersion: v1
+kind: Config
+clusters:
+- name: cluster
+  cluster:
+    certificate-authority: /run/secrets/kubernetes.io/serviceaccount/ca.crt
+    server: https://kubernetes.default
+contexts:
+- name: default
+  context:
+    cluster: cluster
+    user: user
+current-context: default
+users:
+- name: user
+  user:
+    tokenFile: /run/secrets/kubernetes.io/serviceaccount/token
+`
+
+	Namespace = "kube-system"
+	JobName   = "kubectl"
+)
+
+var (
+	importTimeout = int64(60 * 1)
+)
+
+// GetPodAndPodLogs gets the pod name and then capture the podLogs for that pod. 
+//It then returns the podlogs by using the helper GetPodLogs
+func GetPodAndPodLogs(client *rancher.Client, clusterID string) (string, error) {
+	steveClient := client.Steve
```

Comment:
Don't we already have this func?

Reply:
We need the name of the pod for both kubectl apply and kubectl explain function. Instead of duplicating the code to get the pod name, added a function to get the pod name and then calling the existing GetPodLogs function.